source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,140 @@
 Data management
 ===============
 
-The Data Management component includes classes and utilities for data acquisition, initial preprocessing
-and normalization, for data conversion into numeric formats (performed by one of supported Data Sources)
-and for model representation.
+Effective data management is among key constituents of the performance of a data
+analytics application. For |dal_short_name|, effective data management requires
+effectively performing the following operations:
+
+- raw data acquisition
+- filtering and normalization
+- data conversion to a numeric representation
+- data streaming to an algorithm
+
+Key concepts
+============
+
+|dal_short_name| provides a set of customizable interfaces to operate on
+out-of-memory and in-memory data in different usage scenarios, which include
+:term:`batch processing <Batch mode>`, :term:`online processing <Online mode>`,
+and complex scenarios.
+
+Table
```

Comment:
The text above lists the most common ops on the data, acquisition, filtering etc (I'd also group them to highlight the stages, first bullet is about data acquisition, filtering, etc second-conversion to numeric rep, 3d - streaming to the algorithm), all of them are to be described below with respective concepts.
I'd do Key concepts first - it is okay, but I'd move def of the dataset from tables section to it
Then I'd have a section about data sources
Then a section about tables, which starts with general defs/description and introduction to concepts associated with it, etc, then (we need to analyze two options, one is flat in this version of the doc, another-have builder and accessor as subsections of the table section) 
When you discuss/define the table, describe its concept from the perspective of the problems to be solved with it:
- data storage/representation of the heterogeneous (in the general case) data, variety of the formats, dense, sparse, array of structures, etc
- table construction, different scenarios, memory ownership, interaction with data source, user's memory etc
- access to the data that returns the block in required layout  
Use still high level but precise lang. This section would help connect with next (sub)sections on the storage (we  seem to need a subsection on this), builders, accessors. Then sections about e.g. builder will give the additional info/next level (but still high) of details